### PR TITLE
docs: Adds section for funding the contract without receive()

### PR DESCRIPTION
**Pull request by matevz**
- describes a way to fund a contract without triggering the receive() or fallback() functions.
- adds proper subsections to the *Ethereum vs Sapphire* differences section
- adds the Instant finality subsection

[PREVIEW](https://deploy-preview-318--oasisprotocol-sapphire-paratime.netlify.app/dapp/sapphire/guide#override-receive-and-fallback-when-funding-the-contract)

### CODE REVIEW

**Review by CedarMist — APPROVED** [review]
Seems like a good update.

One small improvement I'd suggest is making the differences a bullet-point list at the top of 'Sapphire vs Ethereum', then going into further detail.

It could be noted that transactions are finalized after being mined, with no additional finality window required (so 6 second finality).

Sapphire is a fast, cheap & private Ethereum :)

**Comment by matevz** [comment]
@kostko Can you please check https://github.com/oasisprotocol/sapphire-paratime/pull/318/files#diff-5ba983da752e82d1bbf6d42cc9ea14f59e058ca16e681afac403a48bdb123512R143-R156 for any misinformation I may have written.

File: docs/guide.mdx [hunk 1]
```diff
@@ -74,27 +73,87 @@ ParaTime.
...
+fund the contract's account even though the contract may not implement payable
+`receive()` or `fallback()`!** Or, if these functions do exist, **they will not
+be triggered**. You can send such Oasis-native transactions by using the [Oasis CLI] for example.
+
+[Oasis-native transactions]: https://github.com/oasisprotocol/docs/blob/main/docs/general/manage-tokens/README.mdx
+[Oasis CLI]: https://github.com/oasisprotocol/cli/blob/master/docs/README.md
+
+### Instant Finality
+
+The Oasis Network is a proof of stake network where 2/3 of the validator nodes
```
Comment (kostko) [inline docs/guide.mdx:145]:
```suggestion
The Oasis Network is a proof of stake network where 2/3+ of the validator nodes
```

File: docs/guide.mdx [hunk 2]
```diff
@@ -74,27 +73,87 @@ ParaTime.
...
+need to verify each block in order to consider it final. However, in Ethereum
+the signatures of those validator nodes can be submitted minutes after the block
+is proposed, which makes the block proposal mechanism independent of the
+validation, but adds uncertainty if and when will the proposed block actually be
+finalised.
+
+In the Oasis Network, the 2/3 of **signatures** need to be provided immediately
```
Comment (kostko) [inline docs/guide.mdx:152]:
```suggestion
In the Oasis Network, the 2/3+ of **signatures** need to be provided immediately
```

File: docs/guide.mdx [hunk 3]
```diff
@@ -74,27 +73,87 @@ ParaTime.
...
+after the block is proposed and **the network will halt, until the required
+number signatures are provided**. This means that you can rest assure that any
```
Comment (kostko) [inline docs/guide.mdx:154]:
```suggestion
number signatures are provided**. This means that you can rest assured that any
```

**Review by kostko — APPROVED** [review]
Some minor nits, otherwise the paragraph on "Instant Finality" seems ok.

**Pull request merged**